pkgs/development/python-modules/zulip-emoji-mapping/default.nix

```diff
@@ -0,0 +1,34 @@
+{
+  lib,
+  fetchFromGitHub,
+  python3Packages,
```

Comment:
```suggestion
  setuptools,
```

The reason we don't use `python3Packages` in python-modules is to make it work with multiple python versions.

```
❯ nix-build -A python311Packages.zulip-emoji-mapping
error:
       … in the condition of the assert statement

         at /home/natsukium/src/private/github.com/natsukium/nixpkgs/lib/customisation.nix:365:17:

          364|     in commonAttrs // {
          365|       drvPath = assert condition; drv.drvPath;
             |                 ^
          366|       outPath = assert condition; drv.outPath;

       … in the right operand of the OR (||) operator

         at /home/natsukium/src/private/github.com/natsukium/nixpkgs/pkgs/development/interpreters/python/passthrufun.nix:28:45:

           27|         if lib.isDerivation value then
           28|           lib.extendDerivation (valid value || throw "${name} should use `buildPythonPackage` or `toPythonModule` if it is to be part of the Python packages set.") {} value
             |                                             ^
           29|         else

       (stack trace truncated; use '--show-trace' to show the full trace)

       error: zulip-emoji-mapping should use `buildPythonPackage` or `toPythonModule` if it is to be part of the Python packages set.
```

Reply:
Oh heck

Reply:
May I ask out of personal interest: Does that mean that `python3Packages` always implies the newest python version?

Reply:
The default version (currently python312) is used unless overriden. The update cycle is as follows, basically the version is updated after the release of xx.05.
https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#cpython-update-schedule-python-cpython-update-schedule

That is, the latest version python313 will be released soon, but until nixos 25.05 is released, python3Packages will be an alias for python312Packages.